src/Raven.Client/Http/HttpCache.cs

```diff
@@ -181,6 +183,8 @@ private void FreeSpace()
                     sizeCleared += FreeItem(item);
                 }
             }
+
+            _isFreeSpaceRunning.LowerOrDie();
```

Comment:
This will kill the app if `OrDie` is thrown

Reply:
Replaced all .RaiseOrDie or .LowerOrDie for their not dying alternatives (which was the original behavior in all places anyways). Functions are left there so that people may start implementing with stronger invariants in mind.